ws_api/wealthsimple_api.py

```
@@ -134,9 +57,16 @@ def _bootstrap_device_id_and_client(self) -> None:
-                "https://my.wealthsimple.com/app/login", method="GET", return_response=True
-            )
+            try:
+                headers = {}
+                if WealthsimpleAPI.user_agent:
+                    headers["User-Agent"] = WealthsimpleAPI.user_agent
+
+                resp = self.http.get(
+                    "https://my.wealthsimple.com/app/login", headers=headers
+                )
+            except RequestException as e:
+                raise CurlException(f"HTTP request failed: {e}")
```

Ruff (0.16.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)